src/app/app.cpp

```diff
@@ -169,6 +169,23 @@ namespace atomic_dex
         return disable_coins(coins_to_disable);
     }
 
+    bool
+    application::has_coins_with_balance()
+    {
+        auto* portfolio_page = get_portfolio_page();
+        auto* portfolio_mdl = portfolio_page->get_portfolio();
+        auto portfolio_data = portfolio_mdl->get_underlying_data();
+
+        for (auto& coin : portfolio_data)
```

Comment:
Seems a perfect place to use https://en.cppreference.com/w/cpp/algorithm/all_any_none_of